app/javascript/mastodon/features/public_timeline/components/column_settings.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import ImmutablePropTypes from 'react-immutable-proptypes';
+import { defineMessages, injectIntl, FormattedMessage } from 'react-intl';
+import ColumnCollapsable from '../../../components/column_collapsable';
+import SettingToggle from '../../notifications/components/setting_toggle';
+import SettingText from './setting_text';
+
+const messages = defineMessages({
+  filter_regex: { id: 'public.column_settings.filter_regex', defaultMessage: 'Filter out by regular expressions' },
+  settings: { id: 'public.settings', defaultMessage: 'Column settings' },
+});
+
+class ColumnSettings extends React.PureComponent {
```

Comment:
Since this component is identical for both "public" and "community" columns, I suggest only creating it in one of those, and importing from the other.